src/message_passing.cpp

```diff
@@ -30,4 +30,30 @@ void get_node_comms(MPI_Comm super_comm,
     MPI_Comm_free(sub_comm);
 }
 
+MPI_Datatype define_position_mpi_datatype()
+{
+  MPI_Datatype type;
```

Comment:
I would put this line down immediately before the call to `MPI_Type_create_struct`